packages/data-broker/src/services/dhis/pullers/DataElementsMetadataPuller.ts

```diff
@@ -38,25 +38,26 @@ export class DataElementsMetadataPuller {
     options: PullDataElementsOptions,
   ) => {
     const dataSourcesByDhisType = this.groupDataSourcesByDhisDataType(dataSources);
-    const metadata = [];
+    let metadata: DhisMetadataObject | DataElementMetadata[] = [];
 
     for (const entry of Object.entries(dataSourcesByDhisType)) {
       const [dhisDataType, groupedDataSources] = entry;
       const dataElementCodes = groupedDataSources.map(({ dataElementCode }) => dataElementCode);
       if (dhisDataType === this.dataSourceModel.getDhisDataTypes().INDICATOR) {
         const indicators = await api.fetchIndicators({ dataElementCodes });
-        metadata.push(
-          ...this.translator.translateInboundIndicators(indicators, groupedDataSources),
-        );
+        const results = this.translator.translateInboundIndicators(indicators, groupedDataSources);
+        metadata = metadata.concat(results);
```

Comment:
@biaoli0 since the 'metadata' variable is mutable, for this assignment, could you just do:
`metadata = results`
since we know it will be an empty array?

Reply:
This is within the for loop, which makes it an accumulated results.

Reply:
ah gotcha!